simulator/src/main/java/com/hedera/block/simulator/BlockStreamSimulatorApp.java

```diff
@@ -53,15 +58,73 @@ public BlockStreamSimulatorApp(
             @NonNull Configuration configuration,
             @NonNull BlockStreamManager blockStreamManager,
             @NonNull PublishStreamGrpcClient publishStreamGrpcClient) {
-        this.configuration = configuration;
         this.blockStreamManager = blockStreamManager;
         this.publishStreamGrpcClient = publishStreamGrpcClient;
 
         blockStreamConfig = configuration.getConfigData(BlockStreamConfig.class);
-
         delayBetweenBlockItems = blockStreamConfig.delayBetweenBlockItems();
     }
 
+    /**
+     * Constructor that accepts a custom configuration and initializes dependencies internally.
+     *
+     * @param configuration the custom configuration to use
+     */
+    public BlockStreamSimulatorApp(Configuration configuration) {
+        this(configuration, createComponent(configuration));
+    }
+
+    /**
+     * Private constructor that initializes the class using a configuration and a Dagger component.
+     *
+     * @param configuration the configuration to use
+     * @param component the Dagger component for dependency injection
+     */
+    private BlockStreamSimulatorApp(
+            Configuration configuration, BlockStreamSimulatorInjectionComponent component) {
+
+        this(
+                configuration,
+                component.getBlockStreamManager(),
+                component.getPublishStreamGrpcClient());
+    }
+
+    /**
+     * Default constructor that loads the default configuration and initializes dependencies.
+     *
+     * @throws IOException if an I/O error occurs while loading the default configuration
+     */
+    public BlockStreamSimulatorApp() throws IOException {
+        this(loadDefaultConfiguration());
+    }
```

Comment:
This class is using Dagger DI. Shouldn't the callers just @Inject this as well? Why do we need additional constructors?

Reply:
There can only be one injected constructor. I was thinking about, deleting the other constructors and keeping only the empty one (in terms of passed params). For changing configuration, we could externalize a method to do this at runtime, that would actually aslo benefit as test driver.

Reply:
Ok, so the DI constructor will only be used by the `BlockStreamSimulator` and that's basically using the builder anyway. Maybe we shouldn't use DI for this part?